posydon/popsyn/synthetic_population.py

```diff
@@ -2038,11 +2035,7 @@ def calculate_cosmic_weights(self, SFH_identifier, MODEL_in=None):
         if MODEL_in is None:
             MODEL = DEFAULT_MODEL
         else:
-            for key in MODEL_in:
-                if key not in DEFAULT_MODEL:
-                    raise ValueError(key + " is not a valid parameter name!")
-
-            # write the DEFAULT_MODEL with updates parameters to self.MODEL.
+            # write the DEFAULT_MODEL with updates parameters to MODEL.
             MODEL = DEFAULT_MODEL
```

Comment:
This causes issues with  `Z_max` since it's set to 1.0 in the DEFAULT_MODEL. Adapt

Reply:
`Z_max` set to `None` in `DEFAULT_SFH_MODEL`.